src/js/api/core-loader.js

```diff
@@ -50,12 +50,13 @@ export function requiresPolyfills() {
 export function requiresProvider(model, providerName) {
     const playlist = model.get('playlist');
     if (Array.isArray(playlist) && playlist.length) {
-        const firstSource = Item(playlist[0]).sources[0];
-        if (firstSource) {
+        const sources = Item(playlist[0]).sources;
+        for (let i = 0; i < sources.length; i++) {
```

Comment:
All sources need to be looped through in case unsupported sources are skipped (mpd on iOS).